### PR TITLE
ignore patterns like git

### PR DESCRIPTION
**Pull request by dleviminzi**
This PR contains a small change to match patterns the way that git does when ignoring files to sync. It uses the `pathspec` library which was already one of our deps.

### CODE REVIEW

File: sdk/src/beta9/sync.py [hunk 1]
```diff
@@ -84,15 +90,9 @@ def _read_ignore_file(self) -> list:
         return patterns
 
     def _should_ignore(self, path: str) -> bool:
```
Comment (luke-lombardi) [inline sdk/src/beta9/sync.py:92]: this seems reasonable, can we add a test case?

File: sdk/tests/test_filesyncer.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import os
+from unittest import TestCase
+from unittest.mock import MagicMock
+
+from beta9.sync import IGNORE_FILE_CONTENTS, FileSyncer
+
+
+class TestTaskQueue(TestCase):
```
Comment (luke-lombardi) [inline sdk/tests/test_filesyncer.py:8]: should this be renamed?

**Review by luke-lombardi — CHANGES_REQUESTED** [review]

File: sdk/tests/test_filesyncer.py (hunk 2 above)
Comment (dleviminzi) [reply sdk/tests/test_filesyncer.py:8]: Ohp yeah

File: sdk/tests/test_filesyncer.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+
+class TestFileSyncer(TestCase):
+    def test_init_ignore_file(self):
+        syncer = FileSyncer(gateway_stub=MagicMock())
+        syncer._init_ignore_file()
+        self.assertTrue(syncer.ignore_file_path.exists())
+        os.remove(syncer.ignore_file_path)
+
+    def test_read_ignore_file(self):
```
Comment (luke-lombardi) [inline sdk/tests/test_filesyncer.py:15]: nit-pick but I would clean up the spacing a bit on these tests, it's hard to read. otherwise good to go.

**Review by luke-lombardi — APPROVED** [review]
see last comment, otherwise good to merge

**Pull request merged**